File: backend/app/schemas/review.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ReviewQueueItemResponse(BaseModel):
    """Single item in the PM review queue."""

    id: UUID
    org_id: UUID
    feedback_item_id: UUID
    source_domain: str
    source_company_name: str | None = None
    candidate_customer_id: UUID | None = None
    candidate_customer_name: str | None = None
    candidate_domain: str | None = None
    confidence: float | None = None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ManualMatchRequest(BaseModel):
    """Request body for manual match on a review item."""

    customer_id: UUID = Field(...)

File: frontend/src/types/review.ts
export interface ReviewQueueItem {
  id: string;
  org_id: string;
  feedback_item_id: string;
  source_domain: string;
  source_company_name: string | null;
  candidate_customer_id: string | null;
  candidate_customer_name: string | null;
  candidate_domain: string | null;
  confidence: number | null;
  status: string;
  created_at: string;
}
